components/places/src/storage/bookmarks.rs

```diff
@@ -919,7 +919,14 @@ pub fn bookmarks_get_url_for_keyword(db: &PlacesDb, keyword: &str) -> Result<Opt
     )?;
 
     match bookmark_url {
-        Some(b) => Ok(Some(Url::parse(&b)?)),
+        Some(b) => match Url::parse(&b) {
+            Ok(u) => Ok(Some(u)),
+            Err(e) => {
+                // We don't have the guid to log and the keyword is PII...
+                log::warn!("ignoring invalid url: {:?}", e);
```

Comment:
Can we emit some kind of counter for this, or even record part of the error in Sentry or something? It could be PII, but it would be nice to understand what kinds of syntactically invalid URLs we’re dealing with. Is it just IPs, like the Fenix ticket says, or something more?

That said, I’m not sure what we’d do with that information even if we knew—it’s interesting for my own curiosity, but it wouldn’t explain how they got opened, or written into the database in the first place. We aren’t going to be changing the URL parser or anything.

Reply:
yeah - given what you said and the fact that we don't have anywhere sane to record this to, I'm not going to do anything here. It's something we can revisit once glean is a little closer to us so we don't need to thread it back though android-components etc.